torch/csrc/Exceptions.cpp

```diff
@@ -174,13 +175,25 @@ PyWarningHandler::~PyWarningHandler() noexcept(false) {
         if (source_location.file == nullptr) {
           result = PyErr_WarnEx(PyExc_RuntimeWarning, msg.c_str(), 1);
         } else {
-          result = PyErr_WarnExplicit(
+          std::string file_name{source_location.file};
+          // Sets Python generated warning locations in the attributes
+          const std::string python_suffix{".py"};
```

Comment:
Elaborate on this line - why?

Reply:
Also TODO that WarnExplicit is not ideal since it bypasses the Python warning mechanism (but no easy API)

Reply:
TODO: ideally set attribute, not check .py

Reply:
What is the issue here? That sometimes we get a python file and sometimes a cpp file?
Also can you elaborate on why WarnExplicit is not idea?

Reply:
I'm going to take some additional time to see if we can improve this check by setting an attribute on our warnings. Sometimes we do get a Python file set explicitly by the JIT, probably because it's trying to map Torchscript to Python. WarnExplicit is not ideal because it doesn't respect the Python runtime's (user's) warning filter. WarnExplicit will always bypass the filter.

Reply:
Sounds good. 
Didn't knew about this WarnExplicit behavior...  Potentially we could wrap that file information as "Originally from XXX" where XXX can be either a cpp or python file and always use PyErr_WarnEx?
That way, for jit stuff, you will get both the python of the call site and the python for the original code?

Reply:
An earlier version actually had the behavior but I wasn't happy with how it looked. There are a few other possibilities I want to review. And we now have time to review them since we're not racing into 1.5. 

The current JIT warnings are for the corresponding Python to the Torchscript, and I don't think we want to change that behavior.

Reply:
Ok, let me know when you want more reviews !